LoopKitUI/View Controllers/BasalScheduleTableViewController.swift

```diff
@@ -0,0 +1,396 @@
+                    break
+                }
+            }
+
+            for item in navigationItem.rightBarButtonItems ?? [] {
+                item.isEnabled = !isSyncInProgress
+            }
+
+            navigationItem.hidesBackButton = isSyncInProgress
+        }
+    }
+
+    public var isScheduleValid: Bool {
+        for item in scheduleItems {
+            if !isBasalRateValid(item.value) {
+                return false
+            }
+        }
+        return scheduleItems.count <= maximumScheduleItemCount
+    }
+
+    private func updateTimeLimitsForItemsAdjacentTo(_ index: Int) {
+        let item = scheduleItems[index]
+
+        // Update previous cell's maximumStartTime
+        if index > 0 {
+            let previousIndexPath = IndexPath(row: index-1, section: Section.schedule.rawValue)
+            if let previousCell = tableView.cellForRow(at: previousIndexPath) as? BasalScheduleEntryTableViewCell {
+                previousCell.maximumStartTime = item.startTime - minimumTimeInterval
+            }
+        }
+        // Update next cell's minimumStartTime
+        if index < scheduleItems.count - 1 {
+            let nextIndexPath = IndexPath(row: index+1, section: Section.schedule.rawValue)
+            if let nextCell = tableView.cellForRow(at: nextIndexPath) as? BasalScheduleEntryTableViewCell {
+                nextCell.minimumStartTime = item.startTime + minimumTimeInterval
+                // If next item is now the last item,
+                if index+1 == scheduleItems.count-1 {
+                    nextCell.maximumStartTime = TimeInterval(hours: 24) - minimumTimeInterval
+                }
+            }
+        }
+    }
+
+
+    // MARK: - UITableViewDataSource
+
+    private enum Section: Int {
+        case schedule
+        case sync
+    }
+
+    open override func numberOfSections(in tableView: UITableView) -> Int {
+        if syncSource != nil {
+            return 2
+        }
+
+        return 1
+    }
+
+    open override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section(rawValue: section)! {
+        case .schedule:
+            return scheduleItems.count
+        case .sync:
+            return 1
+        }
+    }
+
+    open override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch Section(rawValue: indexPath.section)! {
+        case .schedule:
+            let cell = tableView.dequeueReusableCell(withIdentifier: BasalScheduleEntryTableViewCell.className, for: indexPath) as! BasalScheduleEntryTableViewCell
+
+            let item = scheduleItems[indexPath.row]
+
+            cell.valueNumberFormatter.minimumFractionDigits = preferredValueFractionDigits()
+            cell.basalRates = allowedBasalRates
+            cell.unitString = unitDisplayString
+            cell.minimumTimeInterval = minimumTimeInterval
+            cell.isReadOnly = isReadOnly || isSyncInProgress
```

Comment:
At several call sites, `isReadOnly` is or-ed with `isSyncInProgress` to describe whether editing is possible. Might be worth adding a computed `var isEditingPossible: Bool { return !isReadOnly && !isSyncInProgress }` to help ensure that disjunction happens.